client/src/components/Board.tsx

```diff
@@ -39,6 +39,7 @@ interface IBoardProps {
 	updateFigures: (figures: IFigure[]) => void;
 	userLocations: IUserLocations;
 	userProfiles: IUserProfiles;
+	setUserProfiles: React.Dispatch<React.SetStateAction<IUserProfiles>>;
```

Comment:
Ideally we don't want to pass down a strict setState function. We want to decouple components as much as possible, board doesnt need to know that it's setting state through the prop, it just needs to know that it can send an update. If we want to change code in App for example to set state conditionally, we'd have to change the type for setUserProfiles